elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreHelper.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Helper for Aggregation Data Store which does the work associated with extracting {@link Query}.
+ */
+public class AggregationDataStoreHelper {
+
+    //TODO Add support for user selected metrics.
+    private static final int AGGREGATION_METHOD_INDEX = 0;
+
+    private Schema schema;
+    private EntityProjection entityProjection;
+    private Set<Dimension> dimensions;
+    private Set<TimeDimension> timeDimensions;
+    private Map<Metric, Class<? extends Aggregation>> metricMap;
+    private FilterExpression whereFilter;
+    private FilterExpression havingFilter;
+
+    public AggregationDataStoreHelper(Schema schema, RequestScope scope) {
+        this.schema = schema;
+        this.entityProjection = scope.getEntityProjection();
+        dimensions = new LinkedHashSet<>();
+        timeDimensions = new LinkedHashSet<>();
+        resolveDimensionLists(dimensions, timeDimensions);
+        Set<Metric> metrics = new LinkedHashSet<>();
+        resolveMetricList(metrics);
+        resolveMetricMap(metrics);
+        splitFilters();
+    }
+
+    /**
+     * @return {@link Query} query object with all the parameters provided by user.
+     */
+    public Query getQuery() {
+        return Query.builder()
+                .schema(schema)
+                .metrics(metricMap)
+                .groupDimensions(dimensions)
+                .timeDimensions(timeDimensions)
+                .whereFilter(whereFilter)
+                .havingFilter(havingFilter)
+                .sorting(entityProjection.getSorting())
+                .pagination(entityProjection.getPagination())
+                .build();
+    }
+
+    /**
+     * Gets whereFilter and havingFilter based on provided filter expression from {@link EntityProjection}.
+     */
+    private void splitFilters() {
+        FilterExpression filterExpression = entityProjection.getFilterExpression();
+        if (filterExpression == null) {
+            whereFilter = null;
+            havingFilter = null;
+            return;
+        }
+        SplitFilterExpressionVisitor visitor = new SplitFilterExpressionVisitor(schema);
+        FilterConstraints constraints = filterExpression.accept(visitor);
+        whereFilter = constraints.getWhereExpression();
+        havingFilter = constraints.getHavingExpression();
+    }
+
+    /**
+     * Gets dimensions and timeDimensions based on relationships and attributes from {@link EntityProjection}.
+     * @param dimensions Empty set of {@link Dimension} objects.
+     * @param timeDimensions Empty set of {@link TimeDimension} objects.
+     */
+    public void resolveDimensionLists(Set<Dimension> dimensions, Set<TimeDimension> timeDimensions) {
+        Set<String> dimensionNames = getRelationships();
+        Set<String> metricNames = getAttributes(); // time dimensions are under attribute in entity projection
```

Comment:
We'd better rename these two to `relationshipNames` and `metricNames` based on your following logic, dimensions can come from both `dimensionNames` and `metricNames`.